os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -496,12 +497,12 @@ void check_heap_corrupt(struct tcb_s *fault_tcb)
 {
 	if (!IS_SECURE_STATE()) {
 #ifdef CONFIG_APP_BINARY_SEPARATION
-		if (IS_FAULT_IN_USER_THREAD(fault_tcb)) {
+		for (int index = 1; index <= CONFIG_NUM_APPS; index++) {
```

Comment:
Instead of hardcoding a value of 1, let us make a definition like below and use it.. 

#ifdef common binary
#define APP_START_IDX 1
#else
#define APP_START_IDX 0
#endif

Please make this change in the header file and also check all the other places where we are hardcoding this value and change it to use this definition. Please make a new commit for it and add to this PR.